test/verification/stratified_couette_flow.jl

```diff
@@ -0,0 +1,264 @@
+using Statistics, Printf
+
+using Oceananigans
+using Oceananigans.TurbulenceClosures
+
+""" Friction velocity squared. See equation (16) of Vreugdenhil & Taylor (2018). """
+function uτ²(model)
+    Δz = model.grid.Δz
+    ν = model.closure.ν
+
+    Up = HorizontalAverage(model, model.velocities.u; frequency=Inf)
+    U = Up(model)[2:end-1]
+
+    uτ²⁺ = ν * abs(U[1] - Uw) / Δz
+    uτ²⁻ = ν * abs(-Uw - U[end]) / Δz
```

Comment:
At first I thought this was a bug --- the velocity prescribed at the bottom wall is `-Uw` though, right? This may deserve a comment. Also due to our k-indexing convention, this looks like you are computing the negative of the gradient and then taking the absolute value. This seems mathematically correct, but is slightly confusing. Perhaps compute the positive gradient and then take the absolute value?